sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala

```diff
@@ -868,7 +868,7 @@ abstract class DynamicPartitionPruningSuiteBase
           |ON f.store_id = s.store_id WHERE s.country = 'DE'
         """.stripMargin)
 
-      checkPartitionPruningPredicate(df, true, false)
+      checkPartitionPruningPredicate(df, false, false)
```

Comment:
@wangyum Why would this PR change this test? This originally can be planned as a subquery duplicate but now it cannot.

Reply:
This test `set SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "-1"`.

Reply:
Please see https://github.com/apache/spark/pull/31984/commits/b510d7da21f8a92af69b1485b72aef6ad5901448#r603143204 for more details.

Reply:
We can add hint to enable DPP:
```scala
    Given("disable broadcast hash join and enable query duplication")
    withSQLConf(SQLConf.DYNAMIC_PARTITION_PRUNING_REUSE_BROADCAST_ONLY.key -> "false",
      SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "-1",
      SQLConf.DYNAMIC_PARTITION_PRUNING_USE_STATS.key -> "true") {
      val df = sql(
        """
          |SELECT /*+ BROADCAST(s) */ f.date_id, f.product_id, f.units_sold, f.store_id FROM fact_stats f
          |JOIN dim_stats s
          |ON f.store_id = s.store_id WHERE s.country = 'DE'
        """.stripMargin)

      checkPartitionPruningPredicate(df, false, true)

      checkAnswer(df,
        Row(1030, 2, 10, 3) ::
        Row(1040, 2, 50, 3) ::
        Row(1050, 2, 50, 3) ::
        Row(1060, 2, 50, 3) :: Nil
      )
    }
```